### PR TITLE
Docs on threading and the event loop

### PR DESCRIPTION
**Pull request by tlambert03**
# Description
Documentation to go with #1210.  Hope this helps to clarify the intended use cases there.  Feedback welcome!

I also added a very basic intro to the event loop, connecting events, and our key/mouse bindings (mostly just copied from https://github.com/napari/napari/issues/1110#issuecomment-609057214).  @kne42 (or anyone!), feel free to add anything you think is relevant there.

(Note: some of the formatting will only make sense if you run `make docs` and view the sphinx docs locally)

### CODE REVIEW

File: docs/source/events/threading.rst
Comment (GenevieveBuckley) [inline docs/source/events/threading.rst:213]: It'd be good to include the imports here too
```python
import napari
import numpy as np
from napari._qt.threading import thread_worker
```

File: docs/source/events/event_loop.rst
Comment (GenevieveBuckley) [inline docs/source/events/event_loop.rst:126]: It would be good to have a short section here with a list of all the available decorators for mouse/keyboard interactions.

**Review by GenevieveBuckley — COMMENTED** [review]
This is great @tlambert03, very clear explanations.

**Comment by tlambert03** [comment]
thank you for reading @GenevieveBuckley!  Will incorporate your suggestions.

File: docs/source/events/threading.rst
Comment (jni) [inline docs/source/events/threading.rst]: fwiw, the only document that ever made sense to me about async is the [trio tutorial](https://trio.readthedocs.io/en/stable/tutorial.html)

File: docs/source/events/threading.rst
Comment (jni) [inline docs/source/events/threading.rst]:
```suggestion
    from napari.qt.threading import thread_worker
```
?

File: docs/source/events/threading.rst
Comment (jni) [inline docs/source/events/threading.rst:69]: WHOA. =D

File: docs/source/events/threading.rst
Comment (tlambert03) [reply docs/source/events/threading.rst]: want me to use that instead?

File: docs/source/events/threading.rst
Comment (tlambert03) [reply docs/source/events/threading.rst]: thanks!
I will also do a little additional cleanup (making sure `:func:` links are working, etc), once #1210 merges.

File: docs/source/events/threading.rst
Comment (jni) [reply docs/source/events/threading.rst]: in addition to. =)

File: docs/source/events/threading.rst
Comment (jni) [inline docs/source/events/threading.rst]:
```suggestion
increased responsiveness in the viewer, you can often save on precious memory
```

File: docs/source/events/threading.rst
Comment (jni) [inline docs/source/events/threading.rst]:
```suggestion
6. Lastly, if the thread total ever goes to "0", we stop the thread by
```

**Review by sofroniewn — APPROVED** [review]
I'm approving, can always edit after release

**Pull request merged**
